src/components/SignupScreen.js

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import { Component } from 'react';
+import { connect } from 'react-redux';
+import {addUser} from '../store/UsersActions';
+
+import {
+  Button,
+  StyleSheet,
+  Text,
+  TextInput,
+  TouchableOpacity,
+  View,
+} from 'react-native';
+
+class SignupScreen extends Component {
+  constructor(){
+    super();
+    this.state = {
+      email: "",
+      password: "",
+      confirmPassword: "",
+    }
+    this.onSignUp = this.onSignUp.bind(this);
+  }
+
+  onSignUp(navigation){
```

Comment:
Vamos remover o **bind** e alteramos a assinatura da função para  `onSignUp = () => {} ` desta maneira a função ja tem acesso ao contexto da classe sem ser preciso fazer bind do this
```suggestion
  }

  onSignUp = (navigation) => {
```